lib/routes/v1.js

```diff
@@ -105,6 +105,15 @@ api.declare({
     return;
   }
 
+  // Figure out which scopes the client needs
+  let requiredScopes = this.WorkerType.scopes(input);
+  // Now check for them if there are any
+  if (requiredScopes.length > 0) {
+    if (!req.satisfies(requiredScopes)) {
```

Comment:
if I'm remembering the code I wrote properly, `requiredScopes` will have that form.

From the later test:
```javascript
+      subject.scopes(wType).should.eql([
+        ['gScope', 'g'],
+        ['rScope1', 'r'],
+        ['rScope2'],
+        ['r'],
+        ['tScope1', 't'],
+        ['tScope2'],
+        ['t'],
+      ]);
```